### ISSUE TITLE
Implement CachedVolume

### ISSUE THREAD

**Issue by constantinpape** [constantinpape/elf#5]
Which cache do we use?
LRU cache?
Need general caching logic.

**Comment by constantinpape** [comment]
Also relevant https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.LRUStoreCache

cc @j6k4m8

**Comment by j6k4m8** [comment]
Yeah, agreed! I think it makes sense if adding another cache layer to stay flexible — it's possible that LRUCache makes sense for some workflows but not others. I'm imagining a "Proactive Cache" that caches the z+1 chunk when you request the zth chunk, or something...
